src/app/market/shared/market.state.ts

```diff
@@ -0,0 +1,70 @@
+import { Injectable } from "@angular/core";
+import {
+	Action,
+	createSelector,
+	State,
+	StateContext,
+	StateToken,
+} from "@ngxs/store";
+import { produce } from "immer";
+import { pluck, tap } from "rxjs/operators";
+
+import { MarketActions } from "./market.actions";
+import { MarketConfig } from "./market.config";
+import { MarketService } from "./market.service";
+import { MarketCurrency, MarketData } from "./market.types";
+
+export interface MarketStateModel {
+	tickers: Record<string, Record<string, MarketData>>;
+}
+
+export const MARKET_STATE_TICKER = new StateToken<MarketStateModel>(
```

Comment:
Hello there!

Why we are defining this if we are using the storage key directly on the state decorator? 🤔

Reply:
Nevermind, just found this approach and the usage into another file